### PR TITLE
simplification of retry policy instantiation

### PR DESCRIPTION
**Pull request by DashaBulanova**

### CODE REVIEW

File: src/Dodo.HttpClient.ResiliencePolicies/RetryPolicy/RetryPolicySettings.cs [hunk 1]
```diff
@@ -1,28 +1,116 @@
...
+		}
+
+		public static RetryPolicySettings Jitter(int retryCount, TimeSpan medianFirstRetryDelay)
+		{
+			return new RetryPolicySettings(SleepDurationProvider.Jitter(retryCount, medianFirstRetryDelay));
+		}
+
+		#region nested class
+
+		private static class SleepDurationProvider
```
Comment (Ceridan) [inline src/Dodo.HttpClient.ResiliencePolicies/RetryPolicy/RetryPolicySettings.cs:79]: Should we leave it here or move to separate file for readability purpose?

**Review by Ceridan — APPROVED** [review]
Look very nice. Settings usage is much cleaner now. Thank you!

File: src/Dodo.HttpClient.ResiliencePolicies/RetryPolicy/RetryPolicySettings.cs (hunk 1 above)
Comment (DashaBulanova) [reply src/Dodo.HttpClient.ResiliencePolicies/RetryPolicy/RetryPolicySettings.cs:79]: I don't have a strong opinion about it... Personally, I don't like the nested classes at all. But! In this case, the one reason why I've had to make it is I can't use language ability to overload static methods and non-static methods. The static one I need to use it in the constructor, non-static - outside the class.

File: src/Dodo.HttpClient.ResiliencePolicies/RetryPolicy/SleepDurationProvider.cs [hunk 2]
```diff
@@ -1,68 +0,0 @@
-using System;
-using System.Collections.Generic;
-using Polly.Contrib.WaitAndRetry;
-
-namespace Dodo.HttpClientResiliencePolicies.RetryPolicy
-{
-	public sealed class SleepDurationProvider : ISleepDurationProvider
```
[the remark this answers is not in the document]
Comment (DashaBulanova) [reply src/Dodo.HttpClient.ResiliencePolicies/RetryPolicy/SleepDurationProvider.cs:7]: thank you, I've just forgotten about it.

**Pull request merged**